src/transformState.js

```diff
@@ -5,7 +5,23 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action in actions) {
+    if (actions[action].type === 'addProperties') {
+      Object.assign(state, actions[action].extraData);
+    } else if (actions[action].type === 'removeProperties') {
+      for (const key of actions[action].keysToRemove) {
+        delete state[key];
+      }
+    } else if (actions[action].type === 'clear') {
+      for (const key in state) {
+        if (state.hasOwnProperty(key)) {
+          delete state[key];
+        }
```

Comment:
Extra `if` statement.
You can iterate over the entire object and remove all fields without `if` statement here